File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/UserList/VIew/Cell/PanoUserListCell.m
//
//  PanoUserListCell.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/16.
//

#import "PanoUserListCell.h"
#import "PanoDefine.h"
#import <Masonry/Masonry.h>

@implementation PanoUserListCell {
    UIImageView *avator;
    UIButton *chooseButton;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        nameLabel = [UILabel new];
        nameLabel.font = fontMax();
        [self.contentView addSubview:nameLabel];
        avator = [UIImageView new];
        avator.image = [UIImage imageNamed:@"room_avator"];
        [self.contentView addSubview:avator];
        
        [avator mas_makeConstraints:^(MASConstraintMaker *make) {
            make.size.mas_equalTo(CGSizeMake(35, 35));
            make.centerY.mas_equalTo(self.contentView);
            make.left.mas_equalTo(self.contentView).offset(defaultMargin);
        }];
        [nameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.right.centerY.mas_equalTo(self.contentView);
            make.height.mas_equalTo(30);
            make.left.mas_equalTo(avator.mas_right).offset(defaultMargin);
        }];
    }
    return self;
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/UserList/View/Cell/PanoUserListCell.swift
//
//  PanoUserListCell.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/11/2.
//

import UIKit

class PanoUserListCell: UITableViewCell {

    var avator: UIImageView!
    
    var nameLabel: UILabel!
    
    var chooseButton: UIButton!
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
//        chooseButton = UIButton(type: .custom)
//        chooseButton.addTarget(self, action: #selector(chooseAction), for: .touchUpInside)
//        chooseButton.setImage(UIImage(named: "userList_unselect"), for: .normal)
//        chooseButton.setImage(UIImage(named: "userList_select"), for: .selected)
//        chooseButton.imageView?.contentMode = .scaleAspectFit
//        self.contentView.addSubview(chooseButton)
        
        nameLabel = UILabel()
        nameLabel.font = FontLarger
        self.contentView.addSubview(nameLabel)
        
        avator = UIImageView()
        avator.image = UIImage(named: "room_avator")
        self.contentView.addSubview(avator)
        
        
//        chooseButton.snp.makeConstraints { (make) in
//            make.size.equalTo(CGSize(width: 30, height: 30))
//            make.centerY.equalToSuperview()
//            make.left.equalToSuperview().offset(defaultMargin)
//        }
        
        avator.snp.makeConstraints { (make) in
            make.size.equalTo(CGSize(width: 35, height: 35))
            make.centerY.equalToSuperview()
            make.left.equalToSuperview().offset(defaultMargin)
        }
        
        nameLabel.snp.makeConstraints { (make) in
            make.right.equalToSuperview()
            make.centerY.equalToSuperview()
            make.left.equalTo(avator.snp.right).offset(defaultMargin)
            make.height.equalTo(30)
        }
        
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
//    @objc func chooseAction() {
//        self.chooseButton.isSelected = !self.chooseButton.isSelected;
//    }
    
}
